contracts/AcceleratingDistributorClaimAndStake.sol

```diff
@@ -0,0 +1,81 @@
+ * only compatible with standard ERC20 tokens, and not tokens that charge fees on transfers, dynamically change
+ * balance, or have double entry-points. It's up to the contract owner to ensure they only add supported tokens.
+ */
+
+contract AcceleratingDistributorClaimAndStake is AcceleratingDistributor {
+    // Contract which rewards tokens to users that they can then stake. MerkleDistributor logic does not impact
+    // this contract at all, but its stored here for convenience to allow claimAndStake to be called by a user to
+    // claim their staking tokens and stake atomically.
+    MerkleDistributorInterface public merkleDistributor;
+
+    /**************************************
+     *               EVENTS               *
+     **************************************/
+
+    event SetMerkleDistributor(address indexed newMerkleDistributor);
+
+    constructor(address _rewardToken) AcceleratingDistributor(_rewardToken) {}
+
+    /**************************************
+     *          ADMIN FUNCTIONS           *
+     **************************************/
+
+    /**
+     * @notice Resets merkle distributor contract called in claimAndStake()
+     * @param _merkleDistributor Address to set merkleDistributor to.
+     */
+    function setMerkleDistributor(address _merkleDistributor) external onlyOwner {
+        merkleDistributor = MerkleDistributorInterface(_merkleDistributor);
+        emit SetMerkleDistributor(_merkleDistributor);
+    }
+
+    /**************************************
+     *          STAKER FUNCTIONS          *
+     **************************************/
+
+    /**
+     * @notice Claim tokens from a MerkleDistributor contract and stake them for rewards.
+     * @dev Will revert if `merkleDistributor` is not set to valid MerkleDistributor contract.
+     * @dev Will revert if any of the claims recipient accounts are not equal to caller, or if any reward token
+     *      for claim is not a valid staking token or are not the same token as the other claims.
+     * @dev Will revert if this contract is not a "whitelisted claimer" on the MerkleDistributor contract.
+     * @dev The caller of this function must approve this contract to spend total amount of stakedToken.
+     * @param claims Claim leaves to retrieve from MerkleDistributor.
+     * @param stakedToken The address of the token to stake.
+     */
+    function claimAndStake(MerkleDistributorInterface.Claim[] memory claims, address stakedToken)
+        external
+        nonReentrant
+        onlyEnabled(stakedToken)
+    {
+        uint256 batchedAmount;
+        uint256 claimCount = claims.length;
+        for (uint256 i = 0; i < claimCount; i++) {
+            MerkleDistributorInterface.Claim memory _claim = claims[i];
+            require(_claim.account == msg.sender, "claim account not caller");
+            require(
+                merkleDistributor.getRewardTokenForWindow(_claim.windowIndex) == stakedToken,
```

Comment:
This is generally how the AcceleratingDistributor works but curious why we want to support multiple different tokens. Why not just limit to one type of token?

Reply:
So to be clear, the `AcceleratingDistributor` has a single `rewardToken` that can be paid out for multiple `stakingTokens`. We want to reward multiple LP positions: WETH-LP, ACX-LP, etc.

The MerkleDistributor can pay out diff tokens per window. We might want to pay out different rewards besides just ACX-LP

Reply:
Alright that makes sense